From 18efc3aeee63388815fd9c3213c38292ddc81dcd Mon Sep 17 00:00:00 2001
From: Seiya Wang <seiya.wang@mediatek.com>
Date: Fri, 24 Apr 2026 11:27:49 +0800
Subject: [PATCH 108/223] ANDROID: ABI: Update the symbol list for mtk

5 function symbol(s) added
  'struct sk_buff* __skb_recv_datagram(struct sock*, struct sk_buff_head*, unsigned int, int*, int*)'
  'void dev_lstats_read(struct net_device*, u64*, u64*)'
  'long vfs_truncate(const struct path*, loff_t)'
  'int vsock_add_tap(struct vsock_tap*)'
  'int vsock_remove_tap(struct vsock_tap*)'

Bug: 505923691
Bug: 506790426
Change-Id: I667d0229d0aff3afe56e70380e40a8009b3b20dd
Signed-off-by: Seiya Wang <seiya.wang@mediatek.com>
(cherry picked from commit 1c5ded243bc242777bc62632c082c086b90ed490)
---
 gki/aarch64/abi.stg     | 93 +++++++++++++++++++++++++++++++++++++++++
 gki/aarch64/symbols/mtk | 23 ++++++----
 2 files changed, 107 insertions(+), 9 deletions(-)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index 326b7296061c..0166679dcc60 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -2813,6 +2813,11 @@ pointer_reference {
   kind: POINTER
   pointee_type_id: 0x349bc1b4
 }
+pointer_reference {
+  id: 0x07bc18d4
+  kind: POINTER
+  pointee_type_id: 0x34b085cc
+}
 pointer_reference {
   id: 0x07c58d49
   kind: POINTER
@@ -351311,6 +351316,17 @@ struct_union {
     member_id: 0xf5debb96
   }
 }
+struct_union {
+  id: 0x34b085cc
+  kind: STRUCT
+  name: "vsock_tap"
+  definition {
+    bytesize: 32
+    member_id: 0xce0b4469
+    member_id: 0x965f12df
+    member_id: 0x7c00e690
+  }
+}
 struct_union {
   id: 0x19d71054
   kind: STRUCT
@@ -388389,6 +388405,13 @@ function {
   parameter_id: 0x397d00ab
   parameter_id: 0x914dbfdc
 }
+function {
+  id: 0x1c823c8e
+  return_type_id: 0x48b5725f
+  parameter_id: 0x32a623d7
+  parameter_id: 0x2e18f543
+  parameter_id: 0x2e18f543
+}
 function {
   id: 0x1c827adb
   return_type_id: 0x48b5725f
@@ -397585,6 +397608,15 @@ function {
   parameter_id: 0x18019702
   parameter_id: 0x4585663f
 }
+function {
+  id: 0x6fa3d22f
+  return_type_id: 0x054f691a
+  parameter_id: 0x1d44326e
+  parameter_id: 0x1f73dead
+  parameter_id: 0x4585663f
+  parameter_id: 0x13580d6c
+  parameter_id: 0x13580d6c
+}
 function {
   id: 0x6fa775f5
   return_type_id: 0x054f691a
@@ -422571,6 +422603,11 @@ function {
   parameter_id: 0x6720d32f
   parameter_id: 0x6720d32f
 }
+function {
+  id: 0x9c6fe51a
+  return_type_id: 0x6720d32f
+  parameter_id: 0x07bc18d4
+}
 function {
   id: 0x9c707fe8
   return_type_id: 0xf435685e
@@ -429894,6 +429931,12 @@ function {
   parameter_id: 0x0483e6f8
   parameter_id: 0xf435685e
 }
+function {
+  id: 0xa91d2e30
+  return_type_id: 0xfc0e1dbd
+  parameter_id: 0x3a8e2ac6
+  parameter_id: 0x27a7c613
+}
 function {
   id: 0xa9432e62
   return_type_id: 0x2d9d2fbb
@@ -439528,6 +439571,15 @@ elf_symbol {
   type_id: 0x9d54e5f2
   full_name: "__skb_pad"
 }
+elf_symbol {
+  id: 0x39692b63
+  name: "__skb_recv_datagram"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x3594896d
+  type_id: 0x6fa3d22f
+  full_name: "__skb_recv_datagram"
+}
 elf_symbol {
   id: 0x7809a362
   name: "__sock_create"
@@ -465253,6 +465305,15 @@ elf_symbol {
   type_id: 0x13488e67
   full_name: "dev_load"
 }
+elf_symbol {
+  id: 0xa052b9b3
+  name: "dev_lstats_read"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x7fd5bd77
+  type_id: 0x1c823c8e
+  full_name: "dev_lstats_read"
+}
 elf_symbol {
   id: 0x9a2dbfca
   name: "dev_mc_add"
@@ -523840,6 +523901,15 @@ elf_symbol {
   type_id: 0x9ea8f2e8
   full_name: "vfs_symlink"
 }
+elf_symbol {
+  id: 0x0796726a
+  name: "vfs_truncate"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xb3e8bd1e
+  type_id: 0xa91d2e30
+  full_name: "vfs_truncate"
+}
 elf_symbol {
   id: 0x2e5511da
   name: "vfs_unlink"
@@ -525217,6 +525287,15 @@ elf_symbol {
   type_id: 0x9f7cdc88
   full_name: "vsnprintf"
 }
+elf_symbol {
+  id: 0xbad4c4e9
+  name: "vsock_add_tap"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x7478d3d6
+  type_id: 0x9c6fe51a
+  full_name: "vsock_add_tap"
+}
 elf_symbol {
   id: 0x24b72fe8
   name: "vsock_addr_bound"
@@ -525289,6 +525368,15 @@ elf_symbol {
   type_id: 0x139ca744
   full_name: "vsock_for_each_connected_socket"
 }
+elf_symbol {
+  id: 0x5ebc8c37
+  name: "vsock_remove_tap"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x7478d3d6
+  type_id: 0x9c6fe51a
+  full_name: "vsock_remove_tap"
+}
 elf_symbol {
   id: 0x0db9531c
   name: "vsprintf"
@@ -527540,6 +527628,7 @@ interface {
   symbol_id: 0x442533a6
   symbol_id: 0x07c5a76f
   symbol_id: 0x55574d11
+  symbol_id: 0x39692b63
   symbol_id: 0x7809a362
   symbol_id: 0x91762fff
   symbol_id: 0x7a4705f8
@@ -530399,6 +530488,7 @@ interface {
   symbol_id: 0x8c0b66d7
   symbol_id: 0x4f142288
   symbol_id: 0x33035bf6
+  symbol_id: 0xa052b9b3
   symbol_id: 0x9a2dbfca
   symbol_id: 0xafae816a
   symbol_id: 0x47c790a6
@@ -536903,6 +536993,7 @@ interface {
   symbol_id: 0x15481b49
   symbol_id: 0xf96f1be3
   symbol_id: 0x355d4901
+  symbol_id: 0x0796726a
   symbol_id: 0x2e5511da
   symbol_id: 0xf33dadbd
   symbol_id: 0x743ba358
@@ -537056,6 +537147,7 @@ interface {
   symbol_id: 0xea7fc751
   symbol_id: 0xd574e01b
   symbol_id: 0x9af4ba91
+  symbol_id: 0xbad4c4e9
   symbol_id: 0x24b72fe8
   symbol_id: 0x74552293
   symbol_id: 0x79291ebd
@@ -537064,6 +537156,7 @@ interface {
   symbol_id: 0x584cd956
   symbol_id: 0x5af22a06
   symbol_id: 0x3d09e223
+  symbol_id: 0x5ebc8c37
   symbol_id: 0x0db9531c
   symbol_id: 0xf29d7680
   symbol_id: 0xbd4af188
diff --git a/gki/aarch64/symbols/mtk b/gki/aarch64/symbols/mtk
index ce28cdee33ba..b0ae9db6f1ec 100644
--- a/gki/aarch64/symbols/mtk
+++ b/gki/aarch64/symbols/mtk
@@ -499,6 +499,7 @@
   dev_kfree_skb_any_reason
   dev_kfree_skb_irq_reason
   dev_load
+  dev_lstats_read
   __devm_add_action
   devm_alloc_etherdev_mqs
   __devm_alloc_percpu
@@ -790,12 +791,12 @@
   drm_atomic_helper_commit_modeset_disables
   drm_atomic_helper_commit_modeset_enables
   drm_atomic_helper_commit_planes
-  __drm_atomic_helper_connector_destroy_state
   drm_atomic_helper_connector_destroy_state
-  __drm_atomic_helper_connector_duplicate_state
+  __drm_atomic_helper_connector_destroy_state
   drm_atomic_helper_connector_duplicate_state
-  __drm_atomic_helper_connector_reset
+  __drm_atomic_helper_connector_duplicate_state
   drm_atomic_helper_connector_reset
+  __drm_atomic_helper_connector_reset
   __drm_atomic_helper_crtc_destroy_state
   __drm_atomic_helper_crtc_duplicate_state
   drm_atomic_helper_damage_iter_init
@@ -1797,8 +1798,8 @@
   __netif_napi_del
   netif_receive_skb
   netif_receive_skb_list
-  __netif_rx
   netif_rx
+  __netif_rx
   netif_set_tso_max_size
   netif_stacked_transfer_operstate
   netif_tx_lock
@@ -2186,8 +2187,8 @@
   pm_genpd_init
   pm_genpd_remove_device
   pm_get_active_wakeup_sources
-  __pm_relax
   pm_relax
+  __pm_relax
   pm_runtime_allow
   __pm_runtime_disable
   pm_runtime_enable
@@ -2202,8 +2203,8 @@
   __pm_runtime_set_status
   __pm_runtime_suspend
   __pm_runtime_use_autosuspend
-  __pm_stay_awake
   pm_stay_awake
+  __pm_stay_awake
   pm_suspend_default_s2idle
   pm_suspend_target_state
   pm_system_wakeup
@@ -2240,8 +2241,8 @@
   proc_dointvec_jiffies
   proc_dointvec_minmax
   proc_doulongvec_minmax
-  _proc_mkdir
   proc_mkdir
+  _proc_mkdir
   proc_mkdir_data
   proc_remove
   proc_set_user
@@ -2672,6 +2673,7 @@
   skb_queue_purge_reason
   skb_queue_tail
   skb_realloc_headroom
+  __skb_recv_datagram
   skb_recv_datagram
   skb_scrub_packet
   skb_set_owner_w
@@ -3686,6 +3688,7 @@
   vfree
   vfs_fsync_range
   vfs_llseek
+  vfs_truncate
   video_devdata
   video_device_alloc
   video_device_release
@@ -3788,14 +3791,16 @@
   vring_transport_features
   vscnprintf
   vsnprintf
+  vsock_addr_bound
+  vsock_addr_cast
   vsock_addr_equals_addr
   vsock_addr_init
-  vsock_addr_cast
-  vsock_addr_bound
+  vsock_add_tap
   vsock_core_register
   vsock_core_unregister
   vsock_deliver_tap
   vsock_for_each_connected_socket
+  vsock_remove_tap
   vsprintf
   vsscanf
   vunmap
-- 
2.50.1

